File: programmers/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = {0,0};
        int yelloww=yellow;
        int yellowh=1;
        answer[0]=yelloww+2;
        answer[1]=yellowh+2;
        for(int i=2;i<yellow;i++){
            if(2*(yelloww+2)+2*yellowh==brown){
                answer[0]=yelloww+2;
                answer[1]=yellowh+2;
                break;
            }
            if(yellow%i==0){
                yelloww=yellow/i;
                yellowh=i;
            }
        }
        if(answer[0]<answer[1]){
            int tmp = answer[0];
            answer[0]=answer[1];
            answer[1]=tmp;
        }
        return answer;
    }
}
/**/

File: programmers/level2/카펫.cpp
#include <cmath>
#include <vector>

using namespace std;

vector<int> solution(int brown, int yellow) {
    vector<int> answer;
    int area = brown + yellow;
    
    for (int height=3; height<=sqrt(area); ++height) {
        if (area % height == 0) {
            int width = area / height;
            if ((width-2)*(height-2) == yellow) {
                answer.push_back(width);
                answer.push_back(height);
                break;
            }
        }
    }
    return answer;
}